extensions/ql-vscode/src/databases/config/db-config-store.ts

```diff
@@ -76,6 +78,84 @@ export class DbConfigStore extends DisposableObject {
     await this.writeConfig(config);
   }
 
+  public async removeDbItem(dbItem: DbItem): Promise<void> {
+    if (!this.config) {
+      throw Error("Cannot remove item if config is not loaded");
+    }
+
+    const config: DbConfig = cloneDbConfig(this.config);
+    const selectedItem: SelectedDbItem | undefined = config.selected;
+
+    // Remove item from databases
+    switch (dbItem.kind) {
```

Comment:
It's possible we initially put those methods in the db-manager because it was easier for writing tests? But I don't have much of an opinion here—@charisk might have a better idea!

Reply:
The existing flow (that is followed by `addNewList` etc.) was intentional and I would say I have a preference for it, but I'm happy to be convinced that this new approach is better. Here's my thinking about the current set up:
- I see the `DbManager` as the entity that coordinates things/deals with domain logic. It interacts with the config store but also knows about other stuff like the expanded state (which I don't see being cleaned up in this PR?), deletion of actual db files (in the future), etc.
- In the `DbConfigStore` we have to provide targeted functions for some stuff (e.g. `doesLocalListExist`) so it seems like having things like `addNewList` fits that pattern nicely.
Happy to have a chat if it helps!